pyswarms/backend/topology/random.py

```diff
@@ -0,0 +1,180 @@
+        clamp : tuple of floats (default is :code:`None`)
+            a tuple of size 2 where the first entry is the minimum velocity
+            and the second entry is the maximum velocity. It
+            sets the limits for velocity clamping.
+
+        Returns
+        -------
+        numpy.ndarray
+            Updated velocity matrix
+        """
+        return ops.compute_velocity(swarm, clamp)
+
+    def compute_position(self, swarm, bounds=None):
+        """Update the position matrix
+
+        This method updates the position matrix given the current position and
+        the velocity. If bounded, it waives updating the position.
+
+        Parameters
+        ----------
+        swarm : pyswarms.backend.swarms.Swarm
+            a Swarm instance
+        bounds : tuple of :code:`np.ndarray` or list (default is :code:`None`)
+            a tuple of size 2 where the first entry is the minimum bound while
+            the second entry is the maximum bound. Each array must be of shape
+            :code:`(dimensions,)`.
+
+        Returns
+        -------
+        numpy.ndarray
+            New position-matrix
+        """
+        return ops.compute_position(swarm, bounds)
+
+    def __compute_neighbors(self, swarm, k):
+        """Helper method to compute the adjacency matrix of the topology
+
+        This method computes the adjacency matrix of the topology using
+        the randomized algorithm proposed in [TSWJ2013]. The resulting
+        topology is a connected graph.
+
+        Parameters
+        ----------
+        swarm : pyswarms.backend.swarms.Swarm
+            a Swarm instance
+        k : int
+            number of neighbors to be considered. Must be a
+            positive integer less than :code:`n_particles-1`
+
+        Returns
+        -------
+        numpy.ndarray
+            Adjacency matrix of the topology
+        """
+        adj_matrix = np.identity(swarm.n_particles, dtype=int)
+        # Generate auxiliary matrix
+        aux_matrix = np.array(
+            [np.random.choice(
+                # Exclude i from the array
+                np.setdiff1d(
+                    np.arange(swarm.n_particles), np.array([i])
+                ), k, replace=False
+            ) for i in range(swarm.n_particles)])
+
+        # Set elements to one
+        adj_matrix[np.arange(swarm.n_particles).reshape(swarm.n_particles, 1), aux_matrix] = 1
+        adj_matrix[aux_matrix, np.arange(swarm.n_particles).reshape(swarm.n_particles, 1)] = 1
+
+        # Compute distance matrix
+        dist_matrix = dijkstra(adj_matrix, directed=False, return_predecessors=False, unweighted=True)
+
+        # Generate connected graph
+        while connected_components(adj_matrix, directed=False, return_labels=False) != 1:
+            for i in range(swarm.n_particles):
+                for j in range(swarm.n_particles):
+                    if dist_matrix[i][j] == 0:
+                        adj_matrix[i][j] = 1
+
```

Comment:
Wanna try `itertools.product` for this nested for-loop? Here's the [documentation](https://docs.python.org/3/library/itertools.html#itertools.product). Sample usage:

```python
import itertools

for i, j in itertools.product(range(swarm.n_particles), repeat=2):
    # Not sure if this ternary operator works, if not, just use the normal one
    adj_matrix[i][j] = 1 if dist_matrix[i][j] == 0

```